librz/bin/bin.c

```diff
@@ -673,6 +673,29 @@ RZ_API RZ_BORROW RzBinMap *rz_bin_object_get_map_at(RZ_NONNULL RzBinObject *o, u
 	return NULL;
 }
 
+/**
+ * \brief Find the binary symbol at offset \p off.
+ *
+ * This function returns the binary symbol that contains offset \p off.
+ * It uses a hashtable for efficient lookup of symbols.
+ *
+ * \param o Reference to the \p RzBinObject instance
+ * \param off Address to search
+ * \param va When false, the offset \p off is considered a physical address; otherwise, a virtual address
+ * \return Pointer to a \p RzBinSymbol containing the address, or NULL if no symbol is found at the address
+ */
+
+RZ_API RZ_BORROW RzBinSymbol *rz_bin_object_get_symbol_at(RZ_NONNULL RzBinObject *o, ut64 off, bool va) {
+    rz_return_val_if_fail(o, NULL);
+    if (!o->symbols) {
+        return NULL;
+    }
+    if (va) {
+        return ht_up_find(o->symbols->virt, off, NULL);
+    }
+    return ht_up_find(o->symbols->phys, off, NULL);
```

Comment:
`o->symbols` is a `RzList`. this is very wrong. the original code was correct, just the way you check for virtual/physical addresses was wrong.

Reply:
Sure. I will make changes soon. Get it done